src/main/java/uk/gov/pay/products/persistence/dao/ProductDao.java

```diff
@@ -17,13 +17,13 @@ protected ProductDao(Provider<EntityManager> entityManager) {
         super(entityManager, ProductEntity.class);
     }
 
-    public Optional<ProductEntity> findByExternalId(String externalId) {
+    public Optional<ProductEntity> findByFriendlyUrlOrExternalId(String friendlyUrlOrExternalId) {
         String query = "SELECT product FROM ProductEntity product " +
-                "WHERE product.externalId = :externalId";
+                "WHERE product.friendlyUrl = :friendlyUrlOrExternalId OR product.externalId = :friendlyUrlOrExternalId";
```

Comment:
We should discuss this as I wonder what should be returned if we have more than one match. We should probably make sure this can never happen, nevertheless should take the best course of action should it happen.

Reply:
It could never happen as both friendlyUrl and external id are unique